browser/speedreader/speedreader_tab_helper.h

```diff
@@ -21,45 +25,17 @@ class SpeedreaderBubbleView;
 // Determines if speedreader should be active for a given top-level navigation.
 class SpeedreaderTabHelper
     : public content::WebContentsObserver,
-      public content::WebContentsUserData<SpeedreaderTabHelper> {
+      public content::WebContentsUserData<SpeedreaderTabHelper>,
+      public SpeedreaderResultDelegate {
  public:
-  enum class DistillState {
-    // Used as an initialization state
-    kUnknown,
-
-    // The web contents is not distilled
-    kNone,
-
-    // Reader mode state that can only be reached when Speedreader is disabled
-    // The Speedreader icon will pop up in the address bar, and the user clicks
-    // it. It runs Speedreader is "Single Shot Mode".  The Speedreader throttle
-    // is created for the following request, then deactivated.
-    //
-    // The first time a user activates reader mode on a page, a bubble drops
-    // down asking them to enable the Speedreader feature for automatic
-    // distillation.
-    kReaderMode,
-
-    // Speedreader is enabled and the page was automatically distilled.
-    kSpeedreaderMode,
-
-    // Speedreader is enabled, but the page was blacklisted by the user.
-    kSpeedreaderOnDisabledPage,
-
-    // Speedreader is disabled, the URL passes the heuristic.
-    kPageProbablyReadable,
-  };
-
-  static bool PageStateIsDistilled(DistillState state) {
-    return state == DistillState::kReaderMode ||
-           state == DistillState::kSpeedreaderMode;
-  }
-
   ~SpeedreaderTabHelper() override;
 
   SpeedreaderTabHelper(const SpeedreaderTabHelper&) = delete;
   SpeedreaderTabHelper& operator=(SpeedreaderTabHelper&) = delete;
 
+  // Helper function to return a weak pointer
```

Comment:
IMO the method purpose is pretty obvious, not sure if a comment like this is useful

Reply:
sure thing. removed it.